monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -369,6 +394,74 @@ object Consumer {
     }
   }
 
+  /** Implementation for [[Consumer.partitionParallel]] */
+  private final class PartitionParallelConsumer[A](nrPartitions: Int, overflowStrategy: OverflowStrategy.Synchronous[A], partition: A => Int, consumer: PartitionedConsumer[A]) extends Consumer[A, Unit] {
+    override def createSubscriber(cb: Callback[Unit], s: Scheduler): (Subscriber[A], AssignableCancelable) = {
+
+      val errors = scala.collection.mutable.ArrayBuffer.empty[Throwable]
+
+      val obs = PublishSubject[A]()
+      val pub = obs.whileBusyBuffer(overflowStrategy)
+      val isUpstreamDone = AtomicBoolean(false)
+
+      val refCount = RefCountCancelable { () =>
+        errors.synchronized {
+          if (errors.nonEmpty) {
+            cb.onError(CompositeException(errors))
+          } else {
+            cb.onSuccess(())
+          }
+        }
+      }
+
+      def subscriber(part: Int): Cancelable = {
+
+        def consumerSubscriber(refID: Cancelable) = new Subscriber[Seq[A]] {
+          implicit def scheduler: Scheduler = s
+
+          def onNext(elements: Seq[A]) =
+            consumer.consume(part, elements).coeval.value match {
```

Comment:
Hi @Fristi, sorry, forgot to return to this.

What do you mean errors don't get propagated? Describe what you expect to have happened.

So do describe how general error handling works:

1. errors being thrown are usually against the contract; we do not have an official way of dealing with errors, because we cannot catch all errors in all cases, or we cannot stream them etc... when exceptions get thrown, the behavior is considered undefined
2. that said we do have to make a reasonable effort to protect against user code, hence all the try/catching
3. in `PublishSubject` this means that when an error gets caught from downstream, the subscriber that generates the error gets unsubscribed and it won't receive any further notifications
4. in `subscribe()` we are also wrapping subscribers (given by users) with `SafeSubscriber` - this has `try/catch` logic in it that also catches errors happening in `onNext` and in case it happens it tries streaming it to `onError` while returning a `Stop` to the source (take a look if interested)

In other words, as long as you use normal `subscribe()` (and not `unsafeSubscribeFn`), your logic should be completely fine. And you can use `unsafeSubscribeFn` in your `def consumer`, but if you do, then you can do the error handling yourself. And in this case I think it is better to do it explicitly, then to rely on the difference between `subscribe()` and `unsafeSubscribeFn`, because the former is for users and has a behavior that is subject to change.